### PR TITLE
os/bluestore: Improve documentation introduced by #57722

### PR DESCRIPTION
**Pull request by anthonyeleven**
os/bluestore: Improve documentation introduced by #57722

This includes what look like some copy/paste errors as well as use of the `confval` directive and removing embedded default values in accordance with the DRY principle.

@aclamk please review carefully in case I have inadvertently introduced errors.

Signed-off-by: Anthony D'Atri <anthonyeleven@users.noreply.github.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: doc/rados/operations/health-checks.rst [hunk 1]
```diff
@@ -851,7 +853,8 @@ To change this, run the following command:
    ceph config set global bdev_stalled_read_warn_lifetime 10
    ceph config set global bdev_stalled_read_warn_threshold 5
 
-this may be done surgically for individual OSDs or a given mask
+this may be done surgically for individual OSDs or a given mask, e.g.
```
Comment (zdover23) [inline doc/rados/operations/health-checks.rst:856]: s/e.g./for example/

I personally like "e.g.", but I think that there are a lot of people who don't know the difference between "e.g." and "i.e." and I don't want to confuse them.

File: doc/rados/operations/health-checks.rst [hunk 2]
```diff
@@ -824,9 +824,10 @@ Or, to disable this alert on a specific OSD, run the following command:
 BLOCK_DEVICE_STALLED_READ_ALERT
 _______________________________
 
-There are certain BlueStore log messages that surface storage drive issues 
+There are BlueStore log messages that surface storage drive issues
```
Comment (zdover23) [inline doc/rados/operations/health-checks.rst:827]: I have a suggestion which might lead to a discussion:

s/surface/reveal/

I think that "surface" sounds jargony and "reveal" sounds more direct.

File: doc/rados/operations/health-checks.rst [hunk 3]
```diff
@@ -863,40 +866,45 @@ this may be done surgically for individual OSDs or a given mask
 WAL_DEVICE_STALLED_READ_ALERT
 _____________________________
 
-A similar warning like ``BLOCK_DEVICE_STALLED_READ_ALERT`` will be raised to
-identify ``stalled read`` instances on a given BlueStore OSD's ``WAL_DEVICE``.
-This warning can be configured via ``bdev_stalled_read_warn_lifetime`` and
-``bdev_stalled_read_warn_threshold`` parameters similarly described in the
+The warning state ``WAL_DEVICE_STALLED_READ_ALERT`` will be raised to
```
Comment (zdover23) [inline doc/rados/operations/health-checks.rst:869]: s/will be/is/

I think that the present tense is easier to read than the future tense (except in certain finicky situations in which you're explaining a series of things that happen one after another, which this is not).

I think that "will be" should be changed to "is" throughout this document wherever doing so doesn't mangle the intended sense of the sentence.

File: doc/rados/operations/health-checks.rst [hunk 4]
```diff
@@ -863,40 +866,45 @@ this may be done surgically for individual OSDs or a given mask
...
 ``BLOCK_DEVICE_STALLED_READ_ALERT`` warning section.
 
 DB_DEVICE_STALLED_READ_ALERT
 ____________________________
 
-A similar warning like ``BLOCK_DEVICE_STALLED_READ_ALERT`` will be raised to
-identify ``stalled read`` instances on a given BlueStore OSD's ``WAL_DEVICE``.
-This warning can be configured via ``bdev_stalled_read_warn_lifetime`` and
-``bdev_stalled_read_warn_threshold`` parameters similarly described in the
+The warning state ``DB_DEVICE_STALLED_READ_ALERT`` will be raised to
```
Comment (zdover23) [inline doc/rados/operations/health-checks.rst:879]: s/will be/is/

File: doc/rados/operations/health-checks.rst [hunk 5]
```diff
@@ -863,40 +866,45 @@ this may be done surgically for individual OSDs or a given mask
...
+This warning can be configured via the :confval:`bdev_stalled_read_warn_lifetime` and
+:confval:`bdev_stalled_read_warn_threshold` options with commands similar to those
+described in the
 ``BLOCK_DEVICE_STALLED_READ_ALERT`` warning section.
 
 BLUESTORE_SLOW_OP_ALERT
 _______________________
 
-There are certain BlueStore log messages that surface storage drive issues 
+There are BlueStore log messages that surface storage drive issues
```
Comment (zdover23) [inline doc/rados/operations/health-checks.rst:889]: s/surface/indicate/

or

s/surface/reveal/

File: doc/rados/operations/health-checks.rst [hunk 6]
```diff
@@ -863,40 +866,45 @@ this may be done surgically for individual OSDs or a given mask
...
+There are BlueStore log messages that surface storage drive issues 
 that can lead to performance degradation and data unavailability or loss.
+When these are seen the storage drive may be failing and should be investigated
+and potentially replaced.
 
 ``log_latency_fn slow operation observed for _txc_committed_kv, latency = 12.028621219s, txc = 0x55a107c30f00``
 ``log_latency_fn slow operation observed for upper_bound, latency = 6.25955s``
 ``log_latency slow operation observed for submit_transaction..``
 
 As there can be false positive ``slow ops`` instances, a mechanism has
```
Comment (zdover23) [inline doc/rados/operations/health-checks.rst:898]: s/As/Because/

File: doc/rados/operations/health-checks.rst [hunk 7]
```diff
@@ -863,40 +866,45 @@ this may be done surgically for individual OSDs or a given mask
...
 been added for more reliability. If in last ``bluestore_slow_ops_warn_lifetime``
```
Comment (zdover23) [inline doc/rados/operations/health-checks.rst:899]: s/If in last/If in the last/

File: doc/rados/operations/health-checks.rst [hunk 8]
```diff
@@ -863,40 +866,45 @@ this may be done surgically for individual OSDs or a given mask
...
+described in the
 ``BLOCK_DEVICE_STALLED_READ_ALERT`` warning section.
 
 BLUESTORE_SLOW_OP_ALERT
 _______________________
 
-There are certain BlueStore log messages that surface storage drive issues 
+There are BlueStore log messages that surface storage drive issues 
 that can lead to performance degradation and data unavailability or loss.
+When these are seen the storage drive may be failing and should be investigated
```
Comment (zdover23) [inline doc/rados/operations/health-checks.rst:891]: s/When these are seen/These indicate that/

File: doc/rados/operations/health-checks.rst [hunk 9]
```diff
@@ -931,17 +939,18 @@ the system. Note that this marking ``out`` is normally done automatically if
...
 To check device health, run the following command:
 
 .. prompt:: bash $
 
    ceph device info <device-id>
 
-Device life expectancy is set either by a prediction model that the Manager
-runs or by an external tool that is activated by running the following command:
+Device life expectancy is set either by a prediction model that the Ceph Manager
+runs or by an external tool by running the following command:
```
Comment (zdover23) [inline doc/rados/operations/health-checks.rst:953]: I think removing the relative clause "that is activated by running the following command" makes this sentence ambiguous. Without the "that", the entire sentence falls under the scope of the prepositional phrase "by running the following command", but that prepositional phrase (I think, if I'm reading this correctly) applies only to the command that is used to designate an external tool as a part of the process of setting the life expectancy of the device. 

TL;DR: I think the sentence should have "that is activated" restored to it.

Comment (anthonyeleven) [reply doc/rados/operations/health-checks.rst:953]: I've pushed all of your most excellent suggestions except for this one.

My sense here is that an external tool may set a value for a given device's life expectancy by running `ceph device set-life-expectancy ...`. As originally written it implies to me that running `ceph device set-life-expectancy ...` _causes_ the external tool to be run / activated.

Comment (zdover23) [reply doc/rados/operations/health-checks.rst]: I'll defer to you. I have no direct experience of this command, so I'm going to regard you as the authority.

**Comment by zdover23** [comment]
jenkins test make check

**Comment by zdover23** [comment]
jenkins test make check arm64

**Comment by zdover23** [comment]
jenkins test make check

**Comment by anthonyeleven** [comment]
Approval please

**Review by zdover23 — APPROVED** [review]

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/60893 - Squid backport
https://github.com/ceph/ceph/pull/60894 - Reef backport
https://github.com/ceph/ceph/pull/60895 - Quincy backport
